sql/core/src/test/scala/org/apache/spark/sql/QueryTest.scala

```diff
@@ -205,6 +206,21 @@ abstract class QueryTest extends PlanTest {
         planWithCaching)
   }
 
+  /**
+   * Asserts that a given [[Dataset]] will be executed using the given named cache.
+   */
+  def assertCached(query: Dataset[_], cachedName: String, storageLevel: StorageLevel): Unit = {
+    val planWithCaching = query.queryExecution.withCachedData
+    val matched = planWithCaching.find( cached =>
```

Comment:
Thanks. I will update it accordingly.

Reply:
Hi @srowen , 

We cannot use `exists { cached => ` here. It is  because planWithCaching is not a `Seq`, but a `LogicalPlan`. It seems that we can traverse it with only methods `collect` or `collectFirst`  

I updated the line to 
```
val matched = planWithCaching.collectFirst { case cached: InMemoryRelation =>
      val cacheBuilder = cached.asInstanceOf[InMemoryRelation].cacheBuilder
      cacheBuilder.tableName.get == cachedName && cacheBuilder.storageLevel == storageLevel
    }.nonEmpty
```
It hope it looks better. 

Thanks and regards,
William